src/algebra/big_operators/intervals.lean

```diff
@@ -38,6 +39,22 @@ lemma prod_Ico_add [ordered_cancel_add_comm_monoid α] [has_exists_add_of_le α]
   (∏ x in Ico a b, f (c + x)) = (∏ x in Ico (a + c) (b + c), f x) :=
 @sum_Ico_add _ (additive β) _ _ _ _ f a b c
 
+lemma sum_range_neg_one_pow [ring β] {n : ℕ} :
+  ∑ i in range n, (-1 : β) ^ i = (n % 2 : ℕ) :=
+begin
+  induction n with n hn,
+  { simp only [range_zero, sum_empty, nat.zero_mod, nat.cast_zero]},
+  simp_rw [sum_range_succ, hn],
+  by_cases even n,
+  { have := (iff_not_comm.mpr nat.even_succ).mp h,
+    rw [nat.even_iff.mp h, nat.not_even_iff.mp this, nat.neg_one_pow_of_even h],
+    norm_num },
+  { have := nat.even_succ.mpr h,
+    rw [nat.even_iff.mp this, nat.not_even_iff.mp h,
+        nat.neg_one_pow_of_odd (nat.odd_iff_not_even.mpr h)],
+    norm_num }
+end
```

Comment:
I think it might be worth breaking out a lemma like this:
```suggestion
lemma coe_nat_succ_mod_two_eq [ring β] {n : ℕ} :
  (↑(n.succ % 2) : β) = ↑(n % 2) + (-1) ^ n :=
begin
  by_cases h : even n,
  { have := (iff_not_comm.mpr nat.even_succ).mp h,
    simp [nat.even_iff.mp h, nat.not_even_iff.mp this, nat.neg_one_pow_of_even h], },
  { have := nat.even_succ.mpr h,
    simp [nat.even_iff.mp this, nat.not_even_iff.mp h,
        nat.neg_one_pow_of_odd (nat.odd_iff_not_even.mpr h)], },
end

lemma sum_range_neg_one_pow [ring β] {n : ℕ} :
  ∑ i in range n, (-1 : β) ^ i = (n % 2 : ℕ) :=
begin
  induction n with n hn,
  { simp, },
  { rw [sum_range_succ, hn, coe_nat_succ_mod_two_eq], }
end
```
and finding an appropriate home for `coe_nat_succ_mod_two_eq`.

Reply:
@ocfnash. Out of curiosity, why do you want this lemma in particular? I'm having a hard time imagining where it would be useful outside the context of this proof.

If this PR cannot be merged without `coe_nat_succ_mod_two_eq`, could you tell me exactly where it belongs in the source tree? It's not exactly `basic`, and while it's related to `parity`, its statement contains neither `even` nor `odd`.

Reply:
I confess that I also cannot quickly think of another application. However I still think this lemma is worth extracting. I think a reasonable home for the lemma would be in `data/nat/parity.lean` [about here](https://github.com/leanprover-community/mathlib/blob/d1ab3dab9244bad78e040b27587f8ecb24195c1c/src/data/nat/parity.lean#L235)